### PR TITLE
Ofi bgq enablement

### PR DESCRIPTION
**Pull request by pkcoff**
test:jenkins/ch4/all
test:jenkins/ch4/ofi

This branch contains the initial changes necessary for the OFI Libfabric implementation on Blue Gene /Q to build and run correctly within MPICH as an OFI provider.

### CODE REVIEW

**Comment by raffenet** [comment]
Thanks for this PR @pkcoff. Only repo admins have the ability to test patches using the trigger phrase, so I'll kick it off for you.

test:jenkins/ch4/all

File: src/mpid/ch4/src/ch4r_symheap.h [hunk 1]
```diff
@@ -66,14 +66,15 @@ static inline int MPIDI_CH4R_check_maprange_ok(void *start, size_t size)
     for (i = 0; i < num_pages; i++) {
         rc = msync(ptr, page_sz, 0);
 
-        if (rc == -1) {
-            MPIR_Assert(errno == ENOMEM);
-            ptr += page_sz;
+        if(rc == -1) {
```
Comment (raffenet) [inline src/mpid/ch4/src/ch4r_symheap.h]: The changes in this file mix whitespace with functional changes, and the whitespace changes do not follow the standard MPICH style. Please fixup the whitespace.

File: src/mpid/ch4/src/ch4r_symheap.h [hunk 2]
```diff
@@ -66,14 +66,15 @@ static inline int MPIDI_CH4R_check_maprange_ok(void *start, size_t size)
...
+            if (errno != ENOMEM) goto fn_fail;
```
Comment (raffenet) [inline src/mpid/ch4/src/ch4r_symheap.h]: put goto on separate, indented line.

File: src/mpid/ch4/src/ch4r_symheap.h [hunk 3]
```diff
@@ -66,14 +66,15 @@ static inline int MPIDI_CH4R_check_maprange_ok(void *start, size_t size)
...
+            ptr+=page_sz;
```
Comment (raffenet) [inline src/mpid/ch4/src/ch4r_symheap.h]: revert whitespace change

File: src/mpid/ch4/src/ch4r_symheap.h [hunk 4]
```diff
@@ -66,14 +66,15 @@ static inline int MPIDI_CH4R_check_maprange_ok(void *start, size_t size)
...
         }
         else
             goto fn_exit;
     }
 
+  fn_fail:
```
Comment (raffenet) [inline src/mpid/ch4/src/ch4r_symheap.h]: `fn_fail` labels typically go below `fn_exit` in MPICH.

File: src/mpid/ch4/netmod/ofi/ofi_impl.h [hunk 5]
```diff
@@ -289,7 +289,7 @@ MPL_STATIC_INLINE_PREFIX MPIDI_OFI_win_request_t *MPIDI_OFI_win_request_alloc_an
 
 MPL_STATIC_INLINE_PREFIX void MPIDI_OFI_win_datatype_unmap(MPIDI_OFI_win_datatype_t * dt)
 {
-    if (dt->map != &dt->__map)
+    if(dt && dt->map && (dt->map != &dt->__map))
```
Comment (raffenet) [inline src/mpid/ch4/netmod/ofi/ofi_impl.h]: revert whitespace change

File: src/mpid/ch4/netmod/ofi/ofi_init.h [hunk 6]
```diff
@@ -306,7 +306,7 @@ static inline int MPIDI_OFI_init_generic(int rank,
     /* vector attribute structure.                                              */
     /* ------------------------------------------------------------------------ */
     char av_name[128];
-    snprintf(av_name, 127, "MPICH-%d\n", appnum);
+    snprintf(av_name, 127, "FI_NAMED_AV_%d\n", appnum);
```
Comment (raffenet) [inline src/mpid/ch4/netmod/ofi/ofi_init.h]: Looks like this should be part of the previous commit.

**Comment by raffenet** [comment]
test:jenkins/ch4/ofi

**Comment by raffenet** [comment]
test:jenkins/ch4/all

**Comment by raffenet** [comment]
I've signed off on a handful of these commits so far. @wesbland is the shared AV patch in here okay with you? @mblockso said he ran it by you but I want to double check before it lands in MPICH.

**Comment by wesbland** [comment]
Yeah, I looked at it yesterday. I think it's fine, though it's not what I originally thought it would be (a shared memory implementation of address vectors. I don't think it prevents the kind of shared memory AV that we want. It's just not the same thing.

**Comment by raffenet** [comment]
Applied.

**Pull request closed without merging**